server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt

```diff
@@ -35,14 +35,18 @@ class ObjectService(
         val objectBytes = obj.toByteArray()
         val objectSize = objectBytes.size.toLong()
 
+        val additionalAudiences = setOf(requesterPublicKey) + additionalAudienceKeys
+
         return objectStoreClient.osClient.put(
             ByteArrayInputStream(objectBytes),
             masterKey.publicKey,
             masterKey.signer(),
             objectSize,
-            setOf(requesterPublicKey),
+            additionalAudiences,
         ).get().hash.toByteArray().base64String().also { hash ->
-            objectPermissionsRepository.addAccessPermission(hash, requesterAddress, objectSize)
```

Comment:
I think we still need this line to ensure the creator can still fetch their stuff

Reply:
However, I'd think at least one test would blow up if the sender couldn't fetch their object anymore.  Maybe our testing around these functions is too sparse?

Reply:
the `additionalAudiences` set does contain the requester's key, so they still get access permission. The verification blocks that verify `objectPermissionsRepository.addAccessPermission(objectHash, ownerKey.getAddress(false), objectBytes.size.toLong())` test that

Reply:
maybe I should rename that value, as it is a little confusing that `additionalAudiences` contains the requester's key + `additionalAudienceKeys`... it is the 'additional audience' from the perspective of object store, but slightly more than that from this perspective

Reply:
done been renam'd

Reply:
Whoops, my bad for not noticing that.  LGTM!